### PR TITLE
Update development and testing tooling

### PR DESCRIPTION
**Pull request by rusterholz**
## What did we change?
A wide range of changes intended to "modernize" the development and test story for this gem:
* Removed @tjwp from the CODEOWNERS, since he is no longer here
* Updated the PR template to the latest standard
* Added standard scripts to facilitate local docker testing (like `docker/setup` and `docker/start`)
* Updated the Dockerfile to use our multiruby image
* Updated the repo's ruby version to 2.7 since multiruby no longer supports ruby 2.6
* Updated to the latest `ezcater_rubocop` and corrected all reported issues
* Updated dev and runtime dependencies:
  * Pinned sidekiq below 6.4.2 for now, to allow our tests to pass
  * Pinned ddtrace below 1.0.0 because that version has breaking changes
  * Removed the bundler dependency since it was clashing with the multiruby image (which has multiple bundlers)
* Substantially refactored how we load database configuration in spec_helper.rb (and renamed several env vars) so that it can support testing in all three common setups: local (e.g. "bare metal"), dockerized, and in Jenkins
* Updated license file to reflect the forward passage of time
* Added a service.yml and Jenkinsfile to support testing this gem in CI
* Updated existing appraisals, commented off the ones that are permanently broken, and added the others as test steps in service.yml.
* Updated the CircleCI config file as follows:
  * Updated to ruby 2.7 (image `cimg/ruby:2.7.6`)
  * Removed references to CodeClimate since we are deprecating that
  * Added missing appraisals for Sidekiq 6.3

## Why are we doing this?
While working on fixing the compatibility issue with sidekiq 6.4.2, I was sometimes slowed down by the lack of some of these things, and thought they might improve the maintainability of this gem going forward.

## Note about Appraisals
The "permanently broken" appraisals are those that use Rails 5.2. That version of Rails only supports versions of ruby up through 2.6, and the lowest ruby version in our multiruby image is 2.7. Unfortunately the specs _do_ break if you attempt to run the Rails 5.2 appraisals on ruby 2.7, in a way that doesn't appear fixable (it looks like they have to do with the changes to kwargs and `**`; ActiveRecord is passing arguments around internally in a way that 2.7 chokes on). We'll have to discuss separately whether we can formally drop support for Rails 5.2 in this gem; if we _don't_ want to do that, then we'll have to change to use an older version of the multiruby image.

## Note about PR Checks
The latest commit on this branch [passed Jenkins testing here](https://jenkins.ezcater.net/blue/organizations/jenkins/ezcater-pipeline%2Fsidekiq_publisher/detail/ar%2Fdevelopment-updates/5/pipeline); however, this repo is not yet configured to require a passing Jenkins build before merging, so no link to Jenkins appears below. If this PR is merged, presumably we can enable that check at that time.

## How was it tested?
- [X] Specs
- [X] Locally
- [ ] Staging

### CODE REVIEW

**Comment by rusterholz** [comment]
@agrobbin I didn't realize that we were actually using CircleCI for some of our repos. I've never seen a CircleCI UI once since joining ezCater... 😂 

Also it seemed to me like it hadn't been run in a while, since it was running rails 5.2 appraisals but the spec_helper.rb file has been [incompatible with rails 5.2 for at least four years](https://github.com/ezcater/sidekiq_publisher/blame/master/spec/spec_helper.rb#L76) (we were passing a URL to `establish_connection` but 5.2 requires hash arguments instead).

Now that I look a little closer, I do see that we've had builds passing in Circle not that long ago (2 months ago was the last non-failed build). So at this point I have no idea how that was _ever_ working... but regardless, I'm happy to put that file back and try to make it align with the service.yml a little better. 👍

File: .circleci/config.yml [hunk 1]
```diff
@@ -24,12 +24,6 @@ jobs:
             - ruby-cache-{{ arch }}-{{ .Branch }}-
             - ruby-cache-
 
-      - run:
-          name: Setup Code Climate test-reporter
-          command: |
-            curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 -o ./cc-test-reporter
-            chmod +x ./cc-test-reporter
```
Comment (jmpage) [inline .circleci/config.yml:31]: This may still report coverage as this is an open source project.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1 @@
+ezcaterPipeline("v1.3")
```
Comment (jmpage) [inline Jenkinsfile:1]: I recommend dropping this as we don't use our internal build pipeline for open source projects.

File: docker/artifactory_setup [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export BUNDLE_EZCATER__JFROG__IO=${BUNDLE_EZCATER__JFROG__IO:-$(grep -e "^BUNDLE_EZCATER__JFROG__IO:" ~/.bundle/config | cut -d\" -f 2)}
```
Comment (jmpage) [inline docker/artifactory_setup:1]: I recommend dropping this file as we don't use private gems for this open source project.

File: service.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
+name: sidekiq-publisher
```
Comment (jmpage) [inline service.yml:1]: I recommend dropping this file as we don't use our internal build pipeline for open source projects.

**Review by jmpage — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (rusterholz) [reply .circleci/config.yml:31]: Does coverage reporting not require a CodeClimate account?

Comment (jmpage) [reply .circleci/config.yml:31]: I believe that it's free for open source: https://codeclimate.com/github/ezcater/sidekiq_publisher

**Pull request closed without merging**